x-pack/legacy/plugins/monitoring/server/routes/api/v1/cluster/cluster.js

```diff
@@ -29,7 +29,8 @@ export function clusterRoute(server) {
           timeRange: Joi.object({
             min: Joi.date().required(),
             max: Joi.date().required()
-          }).required()
+          }).required(),
+          codePaths: Joi.array().items(Joi.string()).required()
```

Comment:
Will this allow empty arrays? If so, I think we should disallow that?

Reply:
Nah, it won't let an empty array go through.


`... --data-binary '{"timeRange":{"min":"2019-08-05T18:52:29.521Z","max":"2019-08-05T19:52:29.521Z"},"codePaths":[""]}'` yields:
```
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "child \"codePaths\" fails because [\"codePaths\" at position 0 fails because [\"0\" is not allowed to be empty]]",
  "validation": {
    "source": "payload",
    "keys": [
      "codePaths.0"
    ]
  }
}
```

Reply:
What about `"codePaths":[]`?

Reply:
That one does not result in an error and just goes through

Reply:
I think we should disallow empty arrays, perhaps by using https://github.com/hapijs/joi/blob/v16.0.0-rc2/API.md#arrayminlimit?

Reply:
I agree, but I think this might also work and might fit more into how we normally do this -> `codePaths: Joi.array().items(Joi.string().required()).required()`

Reply:
That works too!

Reply:
WDYT about changing this a bit and not requiring the array at all? And if it's not present, we assume it's the `ALL` scenario? For one, it'd mean we won't have to update _every_ api call in every test using one :(

Reply:
Correct me if I'm wrong but if we do that doesn't it mean that add a new code path in the future but forgetting to add it's constant, etc. would mean that that new code path would make the "all" call to `GET /clusters` by default?

Reply:
Yea that's the downside. Any new code paths will require explicit knowledge of this to leverage the performance benefits it provides. I'm not sure that's the worst thing though, but happy to hear thoughts from others

Reply:
Yeah, I was stressing on making this Joi schema as strict as possible for this very reason. Without it, we easily risk re-introducing the same bad problem that this PR is trying to resolve. Also happy to hear others thoughts, though.

Reply:
I slept on it and I actually don't like my suggestion. I'm going to leave the validation in and adjust the tests as necessary